src/webview.cpp

```diff
@@ -206,26 +206,69 @@ void WebView::wheelEvent(QWheelEvent *event) {
 
 void WebView::contextMenuEvent(QContextMenuEvent *event)
 {
-    QMenu *menu = new QMenu(this);
-    pageAction(QWebEnginePage::OpenLinkInNewWindow)->setVisible(false);
-    if (!m_linkHovered.isEmpty()) {
-        if (!m_linkHovered.startsWith("zim://")) {
-            pageAction(QWebEnginePage::OpenLinkInNewTab)->setVisible(false);
-            auto openLinkInWebBrowserAction = new QAction(gt("open-link-in-web-browser"));
-            menu->insertAction(pageAction(QWebEnginePage::DownloadLinkToDisk) , openLinkInWebBrowserAction);
-            connect(menu, &QObject::destroyed, openLinkInWebBrowserAction, &QObject::deleteLater);
-            connect(openLinkInWebBrowserAction, &QAction::triggered, this, [=](bool checked) {
-                Q_UNUSED(checked);
-                QDesktopServices::openUrl(m_linkHovered);
-            });
-        } else {
-            pageAction(QWebEnginePage::OpenLinkInNewTab)->setVisible(true);
-        }
+    QMenu* menu;
+    if (m_linkHovered.isEmpty()) {
+        menu = createStandardContextMenu();
+    } else {
+        menu = createLinkContextMenu();
     }
     menu->exec(event->globalPos());
 }
 
 
+QMenu* WebView::createStandardContextMenu() {
+    auto app = KiwixApp::instance();
+
+    QMenu* menu = new QMenu(this);
+    auto backAction = new QAction(gt("back"));
+    backAction->setEnabled(app->getAction(KiwixApp::HistoryBackAction)->isEnabled());
+    backAction->setIcon(app->getAction(KiwixApp::HistoryBackAction)->icon());
+    menu->addAction(backAction);
+    connect(menu, &QObject::destroyed, backAction, &QObject::deleteLater);
+    connect(backAction, &QAction::triggered, this, [=](bool checked) {
+        Q_UNUSED(checked);
+        KiwixApp::instance()->getTabWidget()->triggerWebPageAction(QWebEnginePage::Back);
+    });
+
+    auto forwardAction = new QAction(gt("forward"));
+    forwardAction->setEnabled(app->getAction(KiwixApp::HistoryForwardAction)->isEnabled());
+    forwardAction->setIcon(app->getAction(KiwixApp::HistoryForwardAction)->icon());
+    menu->addAction(forwardAction);
+    connect(menu, &QObject::destroyed, forwardAction, &QObject::deleteLater);
+    connect(forwardAction, &QAction::triggered, this, [=](bool checked) {
+        Q_UNUSED(checked);
+        KiwixApp::instance()->getTabWidget()->triggerWebPageAction(QWebEnginePage::Forward);
+    });
```

Comment:
Can't you use `app->getAction(KiwixApp::History[Back|Forward]Action)` directly instead of creating new temporary `QAction` objects? I think that it should be possible since the handler code doesn't depend on the context (local variables).

Reply:
I was mostly following the previous approach where the `QAction` objects were allocated with `new` and then deleted using `connect`. My understanding is that `menu->addAction` expects the caller to maintain the lifecycle of the object for the duration of the menu's existence. I would guess `HistoryForwardBack` would pretty much always outlive the menu, but I'm not sure. Maybe there are edge cases during shutdown.

Reusing back and forward directly would create an implicit dependency that could cause issues if this ever changed, which might create bugs. And, `HistoryBackAction` is slightly different than `backAction` because it includes a history of items when you long-press (perhaps other differences).

For the savings of a malloc/free and a few bytes of memory, sharing doesn't seem worth the cost IMO.

Reply:
> For the savings of a malloc/free and a few bytes of memory, sharing doesn't seem worth the cost IMO.

I was more concerned about the amount of code (which should always be maintained at a minimum as long as readability is not hurt).

> Reusing back and forward directly would create an implicit dependency that could cause issues if this ever changed, which might create bugs.

There is a dependency in any case and context menus are ephemeral to an extent that I can't imagine a situation causing issues with the implicit dependency without also affecting the explicit one. 

> And, `HistoryBackAction` is slightly different than `backAction` because it includes a history of items when you long-press (perhaps other differences).

That's the strongest point in your argument and it would be enough.